src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ClusterMonitorActor.scala

```diff
@@ -356,10 +356,13 @@ class ClusterMonitorActor(val cluster: Cluster,
       case None => Future.successful(())
       case Some(serviceAccountEmail) =>
         // Only remove the Dataproc Worker role if there are no other clusters with the same owner
-        // in the DB with CREATING status. This prevents situations where we prematurely yank pet SA
-        // roles when the same user is creating multiple clusters.
+        // in the DB with CREATING or UPDATING status. This prevents situations where we prematurely
+        // yank pet SA roles when the same user is creating or resizing multiple clusters.
         dbRef.inTransaction { dataAccess =>
-          dataAccess.clusterQuery.countByClusterServiceAccountAndStatus(serviceAccountEmail, ClusterStatus.Creating)
+          for {
+            creating <- dataAccess.clusterQuery.countByClusterServiceAccountAndStatus(serviceAccountEmail, ClusterStatus.Creating)
```

Comment:
Instead of running 2 queries what if we made `countByClusterServiceAccountAndStatuses` which takes a `Set` of statuses and uses a SQL `IN`? Then we could achieve the same thing with 1 query.